File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.py
from typing import List


class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        row = [1]
        for k in range(1, rowIndex+1):
            row.append(row[-1] * (rowIndex + 1 - k) // k)
        return row


print(Solution().getRow(5))

File: C++/0119-pascals-triangle-ii.cpp
class Solution {
 public:
  vector<int> getRow(int rowIndex) {
    vector<int> curr, prev = {1};

    for (int i = 1; i <= rowIndex; i++) {
      curr.assign(i + 1, 1);

      for (int j = 1; j < i; j++)
        curr[j] = prev[j - 1] + prev[j];

      prev = move(curr);  // This is O(1)
    }

    return prev;
  }
};